polysh/host_syntax.py

```diff
@@ -29,6 +29,12 @@
 syntax_pattern = re.compile('<([0-9,-]+)>')
 interval_pattern = re.compile('([0-9]+)(-[0-9]+)?')
 
+def _split_port(hostname):
+    s = hostname.split(':',1)
+    if len(s) > 1:
+        return s[0], s[1]
+    else:
+        return s[0], '22'
```

Comment:
Another problem is that format strings don't support optional parts so with your format string modification we must always provide a port or ssh will complain about an illegal argument to the `-p` flag. I see two options a) have two format strings or b) find out what port ssh would choose here. I don't like either as polysh currently supports mass-telnet sessions or probably even mass-powershell-remoting if you pass a valid `--ssh` argument to it. It would be really cool to keep this support. Maybe you two have a good idea how to make this pretty?

Reply:
I agree, I didn't think about that particular case. There could be an third option c) Moving the`-p` flag out of the format string and only provide it if necessary. Your workflow should still be functional but it could lead to problems, when telnet/powershell/something is used with this feature: They may not recognise a `-p` flag.

Reply:
That doesn't make me completely happy as telnet takes the port as a positional argument and won't understand -p at all. I could live with that though.